### PR TITLE
expose default cooldown time and hack count

### PR DESCRIPTION
**Pull request by le-jeu**
replace magic numbers by global constants for cooldown and hack count
```js
window.BASE_HACK_COOLDOWN = 300; // 5 mins - 300 seconds
window.BASE_HACK_COUNT = 4;
```
type `COOLDOWN_TIME = 90` in console to get covidown time (on next  portal details render)

### CODE REVIEW

File: core/code/portal_info.js [hunk 1]
```diff
@@ -292,14 +292,14 @@ window.getPortalHackDetails = function(d) {
...
-  var cooldownTime = 300; // 5 mins - 300 seconds 
+  var cooldownTime = COOLDOWN_TIME;
 
   $.each(heatsinks, function(index,mod) {
     var hackSpeed = parseInt(mod.stats.HACK_SPEED)/1000000;
     cooldownTime = Math.round(cooldownTime * (1 - hackSpeed * effectivenessReduction[index]));
   });
 
-  var numHacks = 4; // default hacks
+  var numHacks = HACK_COUNT; // default hacks
```
Comment (johnd0e) [inline core/code/portal_info.js]: Let's keep names in sync.
- Either `var numHacks = NUM_HACKS;`
- Or `var hackCount = HACK_COUNT;`

**Review by johnd0e — CHANGES_REQUESTED** [review]
What else constants we may need?
May be something like `var DOUBLE_AP = false;` or `var AP_FACTOR = 1;`?
What else?

**Comment by le-jeu** [comment]
some times we have double AP on destroyed link, or destroyed field, or both, even triple AP. Also on deploy etc...
The AP gain by action is already exposed tho

**Comment by MysticJay** [comment]
@le-jeu We would need different factors for all actions. we just got double AP for deploying the 8th Reso.

**Comment by johnd0e** [comment]
> We would need different factors for all actions.

Ok then, that is dead end.
I propose to consider custom function for ap gain refinements (not in this PR of cause)

File: core/code/portal_info.js (hunk 1 above)
Comment (MysticJay) [reply core/code/portal_info.js]: 1) AP_FACTOR, not a boolean
2) a COUNT  is something ongoing, better something like MAX_HACKS.

**Comment by MysticJay** [comment]
We should also discuss about a URL, from where these settings can be retrieved.

**Comment by johnd0e** [comment]
> We should also discuss about a URL, from where these settings can be retrieved.

Not in this PR. And even not in core.

**Comment by McBen** [comment]
> What else constants we may need?
> May be something like `var DOUBLE_AP = false;` or `var AP_FACTOR = 1;`?
> What else?

put it into an extra PR

File: core/code/portal_info.js (hunk 1 above)
Comment (McBen) [reply core/code/portal_info.js]: MAX_HACKS sounds good

**Review by johnd0e — APPROVED** [review]

**Review by modos189 — APPROVED** [review]

**Pull request merged**